src/api/ApiService.js

```diff
@@ -0,0 +1,302 @@
+import {
+    collection,
+    deleteDoc,
+    doc,
+    getDoc,
+    getDocs,
+    onSnapshot,
+    query,
+    setDoc,
+    where
+} from 'firebase/firestore';
+import { db } from './Firebase';
+
+class ApiService {
```

Comment:
IMO at this moment functinalities such as post and delete are fine for me (I also don't see anything about adding the informations about user but yeah, it's fine)
We got general setup so I think that for this PR it's fine, rest of the methods should be added in scope of the separate task